File: Code/Visitor.dotNet/Visitor.dotNet/Visitor.dotNet/Visitors/DrawVisitor.cs
using Visitor.Models;

namespace Visitor.Visitors
{
    /// <summary>
    /// A class that draws Rectangles and Squares.
    /// </summary>
    internal class DrawVisitor : IVisitor
    {
        private void NotImplemented()
        {
            Console.WriteLine("I'M SORRY GUYS BUT I'M NOT GOOD AT DRAWING!");
        }

        public void Visit(VisitableSquare s)
        {
            NotImplemented();
        }

        public void Visit(Rectangle s)
        {
            NotImplemented();
        }
    }
}

File: Code/Visitor.Java/src/visitors/DrawVisitor.java
package visitors;

import models.*;

/**
 * A class that draws Rectangles and Squares.
 */
public class DrawVisitor implements IVisitor {

	private void NotImplemented() {
		System.out.println("I'M SORRY GUYS BUT I'M NOT GOOD AT DRAWING!");
	}
	
	@Override
	public void visit(VisitableSquare s) {
		NotImplemented();		
	}

	@Override
	public void visit(Rectangle s) {
		NotImplemented();
	}


}
